packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -482,7 +482,16 @@ class InputDecorator extends StatelessWidget {
       );
     }
 
-    Widget inputChild;
+    Widget inputChild = new KeyedSubtree(
+      // It's important that we maintain the state of our child subtree, as it
+      // may be stateful (e.g. containing text selections). Since our build
+      // function risks changing the depth of the tree, we preserve the subtree
+      // using global keys.
+      // GlobalObjectKey(context) will always be the same whenever we are built.
+      key: new GlobalObjectKey(context),
```

Comment:
That's a clever trick.  Unfortunately, it's a trick with collisions, like the build-in lock in Java's Object.  It's probably fine (no one else is likely to make a global object key out of your context), but if you find yourself using this pattern often, you might consider salting the key with your type or something else to avoid collisions.